kernel/sched.c

```diff
@@ -368,7 +368,16 @@ static void pend(struct k_thread *thread, _wait_q_t *wait_q, s32_t timeout)
 	}
 
 	if (timeout != K_FOREVER) {
-		s32_t ticks = _TICK_ALIGN + k_ms_to_ticks_ceil32(timeout);
+		s32_t ticks;
+
+		__ASSERT(timeout >= 0,
+			"Only non-negative values are accepted.");
+
+		if (timeout < 0) {
+			timeout = 0;
+		}
```

Comment:
What's the value of having both an assert and a runtime clamp?  I'd leave the former in and take this out.  Note that this code gets changed in the timeout patch anyway, where conversion happens in the initialization of the k_timeout_t before we get here.

Reply:
Please removing the the runtime error checking/handling, or remove the assert and handle wrong parameter differently. This is similar to the other PR with k_sleep() and negative values. see https://github.com/zephyrproject-rtos/zephyr/pull/16702 and references in that PR for why.

Reply:
agree with andyross here, @andrewboie  merged while I was commenting, so we will deal with this later when doing the overall runtime checks